config/fcos/v1_6_exp/validate_test.go

```diff
@@ -231,3 +231,123 @@ func TestValidateGrubUser(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateConfig(t *testing.T) {
+	tests := []struct {
+		in      Config
+		out     error
+		errPath path.ContextPath
+	}{
+		// valid config
+		{
+			in: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/vda",
+								WipeTable: util.BoolToPtr(false),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 1,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 2,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		// valid config (wipe_tables is true)
+		{
+			in: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/vda",
+								WipeTable: util.BoolToPtr(true),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 0,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 0,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		// This config specifies the root disk as /dev/disk/by-id/coreos-boot-disk (Valid Config)
+		{
+			in: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/disk/by-id/coreos-boot-disk",
+								WipeTable: util.BoolToPtr(false),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 1,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 2,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		// invalid config, wipetables is false and the partition number is 0
```

Comment:
Maybe something like:

```suggestion
		// invalid config (wipe_table is false with a partition numbered 0)
```